components/cdc/src/channel.rs

```diff
@@ -309,14 +315,15 @@ impl Sink {
         if bytes != 0 && !self.memory_quota.alloc(bytes) {
             return Err(SendError::Congested);
         }
-        match self.unbounded_sender.unbounded_send((event, bytes)) {
-            Ok(_) => Ok(()),
-            Err(e) => {
-                // Free quota if send fails.
-                self.memory_quota.free(bytes);
-                Err(SendError::from(e))
-            }
+        if let Err(e) = self.unbounded_sender.unbounded_send((event, bytes)) {
+            // Free quota if send fails.
+            self.memory_quota.free(bytes);
+            return Err(SendError::from(e));
+        }
+        if let Some(ref limiter) = self.bandwidth_limiter {
+            limiter.blocking_consume(bytes as usize);
```

Comment:
Could you not blocking consume in `unbounded_send` and left a TODO to enable it later?

Real time events are sent by `unbounded_send`, blocking the function may cause OOM.